app.py

```diff
@@ -155,44 +159,36 @@ def callback():
         users_email = userinfo_response.json()["email"]
         picture = userinfo_response.json()["picture"]
         users_name = userinfo_response.json()["name"]
-    else:
-        return "User email not available or not verified by Google.", 400
-
-    # Create a user in our db with the information provided
-    # by Google
-
-    global user
-    user = logUsr.user_info(
-        id_=unique_id, name=users_name, email=users_email, profile_pic=picture
-    )
-    # Doesn't exist? Add to database
-    if not user.get(unique_id):
-        user.create(unique_id, users_name, users_email, picture)
-
-    if "@st.usth.edu.vn" in users_email:
-        # Begin user session by logging the user in
-        login_user(user)
-
-        # Create session timeout
-        time = timedelta(minutes=60)
-        # User will automagically kicked from session after 'time'
-        app.permanent_session_lifetime = time
-
-        # Add user information to Online database
-        id_ = user.get_id()
-        name = user.getName()
-        # temp = name.split()
-        # name = temp[1] + ' ' + temp[2] + ' ' + temp[0]
-        email = user.getEmail()
-        profile_pic = user.getprofile_pic()
-        mongoUsr.register(id_, name, email, profile_pic)
+        if "@st.usth.edu.vn" in users_email or '@usth.edu.vn' in users_email:
+            # Add user information to Online database
+            global user            
+            user = logUsr.user_info(
+                id_=unique_id, name=users_name, email=users_email, profile_pic=picture
+            )
+            id_ = user.get_id()
+            name = user.getName()
+            # temp = name.split()
+            # name = temp[1] + ' ' + temp[2] + ' ' + temp[0]
+            email = user.getEmail()
+            profile_pic = user.getprofile_pic()
+            mongoUsr.register(id_, name, email, profile_pic)
+            generate_password()
+            # Doesn't exist? Add to database
+            if not user.get(unique_id):
```

Comment:
We should check if the account is created bf get their info btw. but these "if statement" to line 168. And put other thing from 168 to 175 to "else statement".